### PR TITLE
WIP

### PR DESCRIPTION
**Pull request by fgelcer**
making the node to be added to the monitor before it starts to bootstrap (to have the bootstrap metrics recorded).
this is due to bootstrap taking ~5 hours issue.

### CODE REVIEW

**Comment by bentsi** [comment]
1) What is WIP? please do not open PRs with such meaningless names
2) Why branch-2019.1? The solution should be merged to master and then backported

**Comment by roydahan** [comment]
He added this PR just to share the code with me.
However, there is an issue here, the function self.check_nodes_up_and_normal is not built for such a long waiting (only 30 retries which is not enough for a node to join).
In this case, there is an exception and the nemesis thread finishes and another node will be decommissioned before the new node was completely added.

**Comment by roydahan** [comment]
Added commit to retest the issue with longer wait, but this whole check needs to be changed.

**Comment by roydahan** [comment]
started this to test https://jenkins.scylladb.com/view/enterprise-2019.1/job/enterprise-2019.1/job/longevity/job/bug_fixes/job/fabio_fix_48h_bootstrap_issue/3/console

**Comment by fruch** [comment]
> He added this PR just to share the code with me.
> However, there is an issue here, the function self.check_nodes_up_and_normal is not built for such a long waiting (only 30 retries which is not enough for a node to join).
> In this case, there is an exception and the nemesis thread finishes and another node will be decommissioned before the new node was completely added.

since it's an open discussion, I agree with @bentsi it should be a bit more clear.
and maybe we could contribute something to that discussion...

**Comment by fruch** [comment]
> He added this PR just to share the code with me.
> However, there is an issue here, the function self.check_nodes_up_and_normal is not built for such a long waiting (only 30 retries which is not enough for a node to join).
> In this case, there is an exception and the nemesis thread finishes and another node will be decommissioned before the new node was completely added.

since it's an open discussion, I agree with @bentsi it should be a bit more clear.
and maybe we could contribute something to that discussion... 

to the point of the change itself, I agree that adding to node to the monitor should take place before node_setup, and in the 5000 tables test, that's exactly what I did. you can see the code there.

**Comment by fgelcer** [comment]
> 1. What is WIP? please do not open PRs with such meaningless names
> 2. Why branch-2019.1? The solution should be merged to master and then backported

exactly to be noticed, and no one merge it by mistake (also, the PR is a DRAFT only).

File: sdcm/cluster.py [hunk 1]
```diff
@@ -2906,7 +2906,7 @@ def check_nodes_up_and_normal(self, nodes=None, verification_node=None):
         if not all(up_statuses):
             raise ClusterNodesNotReady("Not all nodes joined the cluster")
 
-    @retrying(n=30, sleep_time=3, allowed_exceptions=(ClusterNodesNotReady,),
+    @retrying(n=500, sleep_time=20, allowed_exceptions=(ClusterNodesNotReady,),
```
Comment (fgelcer) [inline sdcm/cluster.py]: in that case, it should be ```n=540``` to complete the same expected 3 hours timeout.

**Comment by bentsi** [comment]
How should I understand that it is a discussion and not intended to be merged? Please name the PR properly

**Pull request closed without merging**